--- src/assets/js/theme.main.js	2018-03-21 10:34:18.000000000 +0300
+++ .tmp/assets/js/theme.main.js	2018-05-15 15:30:14.989056600 +0300
@@ -727,4 +727,29 @@
         }
     }
 
+    // R-VISION: RVN-5315
+    $('.external-link').attr('target', '_blank');
+    $('.ht-logo').hide();
+
+    var parent = window.parent;
+    if(parent) {
+        if(parent.R !== undefined && parent.R.theme === 'theme-dark') {
+            $('body, #ht-headerbar, #ht-wrap-container').css({
+                'background-color': '#2b2b2b',
+                color: '#ccc',
+            });
+            $('#ht-sidebar-dragbar, .panel-header').css('background-color', '#101010');
+            $('#ht-sidebar').css('background-color', '#3c3f41');
+            $('.confluenceTh, .confbox').css({
+                'background-color': '#101010',
+                'color': '#ccc',
+            });
+            $('.heading').css('color', '#ccc');
+            $('a').css('color', '#ccc');
+            $('.wiki-content .syntaxhighlighter .plain').css('color', '#ccc');
+            $('head').append('<style> a:hover{ color: #eee !important; } </style>');
+        }
+    }
+    // END 5315
+
 })($);
